src/snakemake/settings/types.py

```diff
@@ -241,6 +242,9 @@ def __post_init__(self):
         if self.remote_job_local_storage_prefix is None:
             self.remote_job_local_storage_prefix = self.local_storage_prefix
 
+        if self.default_storage_provider:
+            self.default_storage_protocol = f"{self.default_storage_provider}://"
```

Comment:
This is dangerous. The name of the provider is not always a valid schema name. We should rather extend the storage provider base class to return the desired schema name via a new abstract method.